File: pallycon-drm-sdk/pallycon_drm_sdk_android/android/src/main/kotlin/com/pallycon/pallycon_drm_sdk_android/DownloadContentHandler.kt
package com.pallycon.pallycon_drm_sdk_android

import DownloadProgressEventImpl
import PallyConEventImpl
import android.content.Context
//import com.google.android.exoplayer2.offline.DownloadService
import com.pallycon.pallycon_drm_sdk_android.sdk.PallyConSdk
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

class DownloadContentHandler: EventChannel.StreamHandler {
    private var channel: EventChannel? = null
    private var context: Context? = null

    fun startListening(context: Context, messenger: BinaryMessenger) {
        if (channel != null) {
            stopListening()
        }

        channel = EventChannel(messenger, "com.pallycon.drmsdk/download_progress")
        channel?.setStreamHandler(this)
        this.context = context

//        try {
//            DownloadService.start(context, DownloadContentService::class.java)
//        } catch (e: IllegalStateException) {
//            DownloadService.startForeground(context, DownloadContentService::class.java)
//        }
    }

    fun stopListening() {
        channel?.setStreamHandler(null)
        channel = null
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        context?.let {
//            PallyConSdk.getInstance(it).setDownloadProgress(events)
            PallyConSdk.getInstance(it).setDownloadProgressEvent(DownloadProgressEventImpl(events))
        }

    }

    override fun onCancel(arguments: Any?) {
        context?.let {
            PallyConSdk.getInstance(it).setDownloadProgressEvent(null)
        }
    }
}

File: pallycon-drm-sdk/pallycon_drm_sdk_ios/ios/Classes/handlers/DownloadContentHandler.swift
import Flutter
import Foundation

public class DownloadContentHandler: NSObject, FlutterStreamHandler {
    private var events: FlutterEventSink?

    public func startListening() {
        
    }

    public func stopListening() {
//        events?.setStreamHandler(null)
//        events = null
    }

    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        // event 연결
        // PallyConSdk.getInstance(it).setDownloadProgress(events)
        self.events = events
        PallyConSdk.shared.setDownloadProgress(eventSink: events)
        return nil
    }

    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        // event 연결 해제
        // PallyConSdk.getInstance(it).setDownloadProgress(null)
        PallyConSdk.shared.setDownloadProgress(eventSink: nil)
        events = nil
        return nil
    }
}

